### DISCUSSION TITLE
Tip and fee design

### DISCUSSION THREAD

**Discussion by taijusanagi** [kaguraswap/monorepo#193] [General]
This is the place to discuss tip and fee degisn

https://miro.com/app/board/uXjVOoCNSfQ=/?moveToWidget=3458764529195465656&cot=14

**Comment by taijusanagi** [comment]
I'm thinking this design

<img width="576" alt="Screen Shot 2022-07-12 at 12 29 51" src="https://user-images.githubusercontent.com/38043569/178402534-866613fc-a4a0-4d6d-90eb-117c2ad943bc.png">

**Reply by taijusanagi** [reply]
The important point is a user can adjust the tip by user choice.

**Reply by taijusanagi** [reply]
The seaport has a tip function so, a tip can be done by fulfiller and offerer. (maybe %can be 2.5% and 2.5% too)

**Reply by taijusanagi** [reply]
0x only accept setting fee at the time of maker-signing, so only the maker can decide the tip

**Reply by ksuhara** [reply]
Maybe we can set the default value like order creator 0% fulfiller 2.5% or order creator 1% fulfiller 1.5%, because we need to chase the number of orders created.

**Comment by ksuhara** [comment]
we can add affiliater for the fulfilling side. maybe not in the initial release though

**Comment by Aqrare** [comment]
The design of a trading fee is nice!
We can add tips to user (Not to BlockBase) like Seaport tip next

**Comment by taijusanagi** [comment]
ok! thank you!
